src/ViewAssertion.php

```diff
@@ -136,7 +136,7 @@ public function has(string $selector): ViewAssertion
     {
         self::assert(function () use ($selector) {
             Assert::assertThat($this->crawler, new CrawlerSelectorExists($selector));
-        }, "Failed asserting that `$selector` exists within %s.");
+        }, "Failed asserting that `{$selector}` exists within `{$this->html}`.");
```

Comment:
@SimoTod I would argue that `self::assert` is now kind of redundant this could just be 

```php
Assert::assertThat(
    $this->crawler,
    new CrawlerSelectorExists($selector), 
    "Failed asserting that `{$selector}` exists within `{$this->html}`."
);
```

And the same in the other assertion methods.

Reply:
@dmason30 
I agree that it's redundant but I don't think we can remove it as part of this patch. 
Despite being private, the ViewAssertion class is 'Macroable' which means that users can add methods at runtime and, in their macro, they can potentially use self::assert.

If I removed that function, all people's macro using it would suddenly break while now the worst thing that could happen is that someone was using %s and it won't replace the placeholder on failure but everything will still be working as expected.

We can mark is as deprecated and remove it when we tag a major release I suppose.

Reply:
Fair point, seems like an appropriate process. Some other packages are less strict ha. I would say that you could still change all the existing assert methods so that it is no longer being used internally and then you could properly  `@deprecate` it and remove in the next major.

Reply:
Yeah, that would work, I'll push an update later on.

It's just that I get really annoyed when I'm on the other side and things break when I run composer update. 😂